packages/smooth_app/lib/pages/scan/continuous_scan_page.dart

```diff
@@ -4,27 +4,75 @@ import 'package:provider/provider.dart';
 import 'package:qr_code_scanner/qr_code_scanner.dart';
 import 'package:smooth_app/data_models/continuous_scan_model.dart';
 import 'package:smooth_app/pages/personalized_ranking_page.dart';
-import 'package:smooth_app/pages/smooth_bottom_navigation_bar.dart';
 import 'package:smooth_app/widgets/smooth_product_carousel.dart';
 import 'package:smooth_ui_library/smooth_ui_library.dart';
 import 'package:smooth_ui_library/util/ui_helpers.dart';
 
-class ContinuousScanPage extends StatelessWidget {
+class ContinuousScanPage extends StatefulWidget {
+  const ContinuousScanPage({required this.routeObserver});
+  final RouteObserver<ModalRoute<void>> routeObserver;
+
+  @override
+  State<ContinuousScanPage> createState() => _ContinuousScanPageState();
+}
+
+class _ContinuousScanPageState extends State<ContinuousScanPage>
+    with RouteAware {
   final GlobalKey _scannerViewKey = GlobalKey(debugLabel: 'Barcode Scanner');
+  ContinuousScanModel? model;
+  //isCurrent (latest) route in navigator stack, tracked to decide whether to rebuild scanner or just to reactivate the camera.
+  bool isCurrent = true;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    widget.routeObserver.subscribe(this, ModalRoute.of(context)!);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.routeObserver.unsubscribe(this);
+  }
+
+  @override
+  void didPushNext() {
+    super.didPushNext();
+    if (model != null) {
+      model!.stopQRView();
+    }
+  }
+
+  //Called when the top route has been popped off, so restarting camera
+  @override
+  void didPopNext() {
+    super.didPopNext();
+    if (!isCurrent) {
+      isCurrent = true;
+      //First building of scanner
+      setState(() {});
+    } else if (model != null) {
+      model!.restartQRView();
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
+    if (!ModalRoute.of(context)!.isCurrent) {
+      isCurrent = false;
+      return const Center(child: Text('A unknown error occurred'));
```

Comment:
A bit fishy.

Reply:
I still think there's a flaw if you have to check twice, here and in ScanPage: `if (widget.offstage) ...`.